share/doc/homebrew/Python-for-Formula-Authors.md

```diff
@@ -54,40 +54,41 @@ Formulæ for apps that require Python 3 **should** declare an unconditional depe
 
-Applications should be installed to `libexec`. This prevents the app's Python modules from contaminating the system site-packages, which is important so that pip doesn't try to manage Homebrew-installed packages and because applications' Python dependencies should not be installed to an importable prefix (see below) so `import` won't work anyway.
+Applications should be installed into a Python [virtualenv](https://virtualenv.pypa.io/en/stable/) environment rooted in `libexec`. This prevents the app's Python modules from contaminating the system site-packages and vice versa.
 
-In your formula's `install` method, first set the `PYTHONPATH` environment variable to your package's libexec site-packages directory with:
-```ruby
-ENV.prepend_create_path "PYTHONPATH", libexec/"lib/python2.7/site-packages"
-```
-Then, use `system` with `Language::Python.setup_install_args` to invoke `setup.py` like:
-```ruby
-system "python", *Language::Python.setup_install_args(libexec)
-```
-
-This will have placed the scripts your Python package installs in `libexec/"bin"`, which is not symlinked into Homebrew's prefix. We need to make sure these are installed and we also need to make sure that, when they are invoked, `PYTHONPATH` includes the path where we just installed your package. Do this with:
-
-```ruby
-bin.install Dir[libexec/"bin/*"]
-bin.env_script_all_files(libexec/"bin", :PYTHONPATH => ENV["PYTHONPATH"])
-```
-The first line copies all of the executables to bin. The second line writes stubs to bin that set `PYTHONPATH` and moves the original files back to `libexec/"bin"`.
-
-## Dependencies
+All of the Python module dependencies of the application (and their dependencies, recursively) should be declared as `Resource`s in the formula and installed into the virtualenv, as well. Each dependency should be explicitly specified; please do not rely on setup.py or pip to perform automatic dependency resolution, for the [reasons described here](Acceptable-Formulae.md#we-dont-like-install-scripts-that-download-things).
 
-All Python dependencies of applications that are not packaged by Homebrew (and those dependencies' Python dependencies, recursively) **should** be unconditionally downloaded as `Resource`s and installed into the application keg's `libexec/"vendor"` path. This prevents the state of the system Python packages from being affected by installing an app with Homebrew and guarantees that apps use versions of their dependencies that are known to work together. `libexec/"vendor"` is preferred to `libexec` so that formulæ don't accidentally install executables belonging to their dependencies, which can cause linking conflicts.
+You can use [homebrew-pypi-poet](https://pypi.python.org/pypi/homebrew-pypi-poet) to help you write resource stanzas. To use it, set up a virtualenv and install your package and all its dependencies. Then, `pip install homebrew-pypi-poet` into the same virtualenv. Running `poet foo` will generate the necessary resource stanzas.
```

Comment:
Concerns me a little that e.g. I would have no idea how to create a virtualenv and install my package and dependencies. Could we maybe provide a helper script/command for this?